plugins/linux/ifplugin/linuxcalls/link_linuxcalls.go

```diff
@@ -232,3 +232,49 @@ func (h *NetLinkHandler) AddVethInterfacePair(ifName, peerIfName string) error {
 	}
 	return nil
 }
+
+// AddVRFDevice configures new VRF network device.
+func (h *NetLinkHandler) AddVRFDevice(vrfDevName string, routingTable uint32) error {
+	attrs := netlink.NewLinkAttrs()
+	attrs.Name = vrfDevName
+	link := &netlink.Vrf{
+		LinkAttrs: attrs,
+		Table:     routingTable,
+	}
+	if err := netlink.LinkAdd(link); err != nil {
+		return errors.Wrapf(err, "failed to add VRF device: LinkAdd (vrf=%s, rt=%d)",
```

Comment:
The part with `failed to add VRF device:` should not be added here, because it is already [added by the caller](https://github.com/milanlenco/vpp-agent/blob/1a3d4fc48832cb02dc3ad3bc47abe5b024349155/plugins/linux/ifplugin/descriptor/interface_vrfdev.go#L55-L59), so it is redundant and would make the error look like:

```
...: creating X interface failed: failed to add VRF device X (rt: Z): failed to add VRF device: LinkAdd (vrf=Y, rt=Z)
```


The errors returned to caller should only add error context that caller does not know about. In this case "failed to **add VRF device**" is not needed because caller knows it since it called `AddVRFDevice` function. What it does not know that this function failed during call to `LinkAdd` so that part is adding useful context info to the error.